blog/urls.py

```diff
@@ -0,0 +1,10 @@
+from django.urls import path, include
+
+from blog.views import index, PostDetailView
+
+urlpatterns = [
+    path("", index, name="index"),
+    path("posts/<int:pk>/", PostDetailView.as_view(), name="post-detail"),
```

Comment:
a bit strange that all your posts can be retrieved by "localhost:8000" url, but specific post by "localhost:8000/posts/id/", don't you think so?
By the best practices, we create such urls:
posts/ - get all posts
posts/id/ - get specific post
posts/update/ | delete/